internal/moreos/testdata/fake_func-e.go

```diff
@@ -31,23 +32,41 @@ func main() {
 		os.Exit(1)
 	}
 
+	// This is similar to main.go, except we don't import the validation error
+	if err := run(context.Background(), os.Args[2:]); err != nil {
+		moreos.Fprintf(os.Stderr, "error: %s\n", err) //nolint
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+// simulates envoy.Run with slight adjustments
+func run(ctx context.Context, args []string) (err error) { //nolint:gocyclo
 	// Like envoy.GetHomeVersion, $FUNC_E_HOME/versions/$(cat $FUNC_E_HOME/version)/bin/envoy$GOEXE.
-	cmd := exec.Command(os.Getenv("ENVOY_PATH"), os.Args[2:]...)
+	cmd := exec.Command(os.Getenv("ENVOY_PATH"), args...)
 	cmd.SysProcAttr = moreos.ProcessGroupAttr()
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 
-	// Like envoy.Run.
-	waitCtx, waitCancel := context.WithCancel(context.Background())
+	// Suppress any error and replace it with the envoy exit status when > 1
+	defer func() {
+		if cmd.ProcessState != nil && cmd.ProcessState.ExitCode() > 0 {
```

Comment:
ExitCode() is safe to call even when ProcessState == nil and in that case it returns `-1`
```suggestion
		if cmd.ProcessState.ExitCode() > 0 {
```

Reply:
wow sneaky nil goodness!

```Golang
// ExitCode returns the exit code of the exited process, or -1
// if the process hasn't exited or was terminated by a signal.
func (p *ProcessState) ExitCode() int {
	// return -1 if the process hasn't started.
	if p == nil {
		return -1
	}
	return p.status.ExitStatus()
}
```

Reply:
fixed this in read func-e also